src/payment/strategies/moneris/moneris-payment-strategy.ts

```diff
@@ -0,0 +1,135 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError , MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { StoreCreditActionCreator } from '../../../store-credit';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentInitializeOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import MonerisStylingProps,  { MonerisResponseData } from './moneris';
+
+const IFRAME_NAME = 'moneris-payment-iframe';
+const RESPONSE_SUCCESS_CODE = '001';
+
+export default class MonerisPaymentStrategy implements PaymentStrategy {
+    private _iframe?: HTMLIFrameElement;
+    private _monerisURL?: string;
+    private _windowEventListener?: (response: MessageEvent) => void;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _storeCreditActionCreator: StoreCreditActionCreator
+    ) {}
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const state = this._store.getState();
+
+        const { moneris: monerisOptions } = options;
+        const { config, initializationData } = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+
+        if (!monerisOptions) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.moneris" argument is not provided.');
+        }
+
+        if (!initializationData?.profileId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this._iframe = this._createIframe(monerisOptions.containerId, initializationData.profileId, !!config.testMode);
```

Comment:
maybe you can check that the iframe hasn't been created?


if (this._iframe) {
 throw new Error('Already initalized');
}

?